tests/test_base.py

```diff
@@ -212,3 +216,90 @@ def test_delete_all_actions():
         assert (
             len(found_actions) == 0
         ), f"Action {action_id} was not deleted from the database"
+
+def test_action_event_with_opts_and_reviews():
+    # Create ActionEvent with ActionOpts
+    state = EnvState(text="Initial state of the environment")
+    tool = V1ToolRef(module="tool_module", type="tool_type", version="1.0")
+
+    owner_id = "owner123"
+
+    action = V1Action(name="test_action", parameters={"param1": "value1"})
+    action_event = ActionEvent(
+        state=state,
+        action=action,
+        tool=tool,
+        namespace="default",
+        owner_id=owner_id,
+    )
+
+    action_event.add_actionOpt(action=V1Action(name="opt1", parameters={"key1": "val1"}))
+    action_event.add_actionOpt(action=V1Action(name="opt2", parameters={"key2": "val2"}))
+
+    # Step 3: Fetch the ActionEvent using the find method with owner_id and verify ActionOpts load correctly
+    found_actions = ActionEvent.find(id=action_event.id)
+    assert len(found_actions) == 1
+
+    found_action_event = found_actions[0]
+    assert found_action_event.id == action_event.id
+    assert len(found_action_event.action_opts) == 2
+    print("ActionEvent and ActionOpts saved and loaded correctly.")
+
+    # Step 4: Use ActionOpt find method to retrieve ActionOpts from the database by action_id
+    found_action_opts = ActionOpt.find(action_id=action_event.id)
+    assert len(found_action_opts) == 2
+    print("ActionOpts found in the database.")
+
+    # Step 5: Add Reviews to the ActionOpts
+    review1 = Review(
```

Comment:
The actions somehow need to also support numerical ratings, the basic idea is that each action is rated on a scale of 1-5, the scale may vary though so that should be configurable too